permhandler/permhandler.go

```diff
@@ -157,12 +159,12 @@ func generateSecret() (secret secretKey, err error) {
 	return secretKey(base64.StdEncoding.EncodeToString(b)), nil
 }
 
-func (handler *Handler) findServiceID(serviceID string) (secret string, err error) {
+func (handler *Handler) getSecretForInstance(instance cloudprotocol.InstanceIdent) (string, error) {
 	for key, value := range handler.secrets {
-		if value.serviceID == serviceID {
+		if reflect.DeepEqual(value.instaneIdent, instance) {
```

Comment:
you can compare instances like variables without using reflect.DeepEqual
```
if value.instaneIdent = instance {
```

Reply:
done, thanks